src/libraries/System.Threading.Tasks.Dataflow/tests/Dataflow/DataflowBlockExtensionTests.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,278 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Threading.Tasks.Dataflow.Tests
+{
+    public partial class DataflowBlockExtensionTests
+    {
+        [Fact]
+        public void ReceiveAllAsync_ArgumentValidation()
+        {
+            Assert.Throws<ArgumentNullException>(() => ((IReceivableSourceBlock<int>)null).ReceiveAllAsync());
+            Assert.Throws<ArgumentNullException>(() => ((IReceivableSourceBlock<int>)null).ReceiveAllAsync(new CancellationToken(true)));
+        }
+
+        [Fact]
+        public void ReceiveAllAsync_NotIdempotent()
+        {
+            var source = new BufferBlock<int>();
+            IAsyncEnumerable<int> e = source.ReceiveAllAsync();
+            Assert.NotNull(e);
+            Assert.NotSame(e, source.ReceiveAllAsync());
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task ReceiveAllAsync_UseMoveNextAsyncAfterCompleted_ReturnsFalse(bool completeWhilePending)
+        {
+            var source = new BufferBlock<int>();
+            IAsyncEnumerator<int> e = source.ReceiveAllAsync().GetAsyncEnumerator();
+
+            ValueTask<bool> vt;
+            if (completeWhilePending)
+            {
+                source.Complete();
+                vt = e.MoveNextAsync();
+                Assert.False(await vt);
+                Assert.True(vt.IsCompletedSuccessfully);
```

Comment:
Operations like this are not safe.  You can't touch `vt` after you've awaited it.

Reply:
OK. I removed this part of the test.